diff --git a/Server/post/views.py b/Server/post/views.py
index 21561c8..f5943d0 100644
--- a/Server/post/views.py
+++ b/Server/post/views.py
@@ -27,9 +27,11 @@ def post_single(request, pk):
 
 def post_add(request):
     if request.method == "POST":
-        form = PostForm(request.POST, request.FILES, instance=Post)
+        form = PostForm(request.POST, request.FILES)
         if form.is_valid():
-            form.save()
+            post = form.instance
+            post.user = request.user
+            post.save()
             return HttpResponseRedirect('/')
     else:
         form = PostForm()
